pkg/bpf/bpf_linux.go

```diff
@@ -113,7 +115,29 @@ func UpdateElementFromPointers(fd int, structPtr unsafe.Pointer, sizeOfStruct ui
 	}
 
 	if ret != 0 || err != 0 {
-		return fmt.Errorf("Unable to update element for map with file descriptor %d: %s", fd, err)
+
+		var errMsg error
+
+		switch err {
+		case unix.E2BIG:
+			if strings.Contains(mapName, "lb") {
+				errMsg = errors.New("the bpf LB map is full, please consider resizing it with the flag \"bpf-lb-max-entries\". THIS WILL INTERFERE WITH ALL EXISTING CONNECTIONS")
+			} else {
+				errMsg = errors.New("the bpf map is full, please consider resizing it")
+			}
+		case unix.EEXIST:
+			errMsg = errors.New("Cannot add the element to bpf map, please check the key of the element")
+		case unix.ENOENT:
+			errMsg = errors.New("Cannot add/update element to bpf map, the element doesn't exist")
```

Comment:
According the the `BPF_MAP_UPDATE_ELEM` section of `bpf(2)`:

> On  success, the operation returns zero.  On error, -1 is returned and errno is set to EINVAL, EPERM, ENOMEM, or E2BIG.  E2BIG indicates that the number of elements in the  map reached the max_entries limit specified at map creation time.  EEXIST will be returned if flags specifies BPF_NOEXIST and the element with key already  exists in  the  map.  ENOENT will be returned if flags specifies BPF_EXIST and the element with key doesn't exist in the map.

I would suggest rephrasing the above message as  "failed to update bpf map %s: key does not exist"